src/activity_handlers.cpp

```diff
@@ -295,27 +298,28 @@ void set_up_butchery( player_activity &act, player &u, butcher_type action )
     }
     // workshop butchery (full) prequisites
     if( action == BUTCHER_FULL ) {
-            if( !g->m.has_flag_furn( "BUTCHER_EQ", u.pos() ) ) {
-                u.add_msg_if_player( m_info, _( "You need a butchering rack to perform a full butchery." ) );
+            if( corpse_item.get_mtype()->size >= MS_MEDIUM && !g->m.has_flag_furn( "BUTCHER_EQ", u.pos() ) ) {
+                u.add_msg_if_player( m_info, _( "For a corpse this big you need a butchering rack to perform a full butchery." ) );
                 act.set_to_null();
                 return;
             }
-            if ( !has_table_nearby ) {
-                u.add_msg_if_player( m_info, _( "You need a table nearby or something else with a flat surface to perform a full butchery." ) );
+            if ( corpse_item.get_mtype()->size >= MS_MEDIUM && !has_table_nearby ) {
```

Comment:
`corpse_item.get_mtype()->size` can be changed to `corpse->size`, or `corpse.size` if changed to a const reference.